packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx

```diff
@@ -25,26 +29,91 @@ type Props = {
     anchorElement?: HTMLElement;
     /** Where should the bubble try to go with respect to its anchor. */
     placement: Placement;
+    /**
+     * Whether the tooltip should automatically update its position when the
+     * anchor element changes.
+     */
+    autoUpdate?: boolean;
 };
 
-export default class TooltipPopper extends React.Component<Props> {
-    _bubbleRefTracker: RefTracker = new RefTracker();
-    _tailRefTracker: RefTracker = new RefTracker();
+/**
+ * A ref tracker for the bubble element.
+ */
+const _bubbleRefTracker: RefTracker = new RefTracker();
+/**
+ * A ref tracker for the tail element.
+ */
+const _tailRefTracker: RefTracker = new RefTracker();
+/**
+ * A MutationObserver that watches for changes to the anchor element.
+ */
+let _observer: MutationObserver | null = null;
+/**
+ * A function that can be called to update the popper.
+ * @see https://popper.js.org/docs/v2/lifecycle/#manual-update
+ */
+let _popperUpdate: PopperUpdateFn | null = null;
+
+/**
+ * A component that wraps react-popper's Popper component to provide a
+ * consistent interface for positioning floating elements.
+ */
+export default function TooltipPopper(props: Props): React.ReactElement {
+    const isMounted = useIsMounted();
+    const {anchorElement, autoUpdate, placement} = props;
 
-    _renderPositionedContent(
+    /**
+     * Automatically updates the position of the floating element when necessary
+     * to ensure it stays anchored.
+     *
+     * NOTE: This is a temporary solution that checks for changes in the anchor
+     * element's DOM. This is not a perfect solution and may not work in all
+     * cases. It is recommended to use the autoUpdate prop only when necessary.
+     *
+     * TODO(WB-1680): Replace this with floating-ui's autoUpdate feature.
+     * @see https://floating-ui.com/docs/autoupdate
+     */
+    React.useEffect(() => {
+        if (!anchorElement || !autoUpdate) {
+            return;
+        }
+
+        if (isMounted()) {
+            _observer = new MutationObserver(async () => {
+                // Update the popper when the anchor element changes.
+                await _popperUpdate?.();
+            });
+
+            // Check for DOM changes to the anchor element.
+            _observer.observe(anchorElement, {
+                attributes: true,
+                childList: true,
+                subtree: true,
+            });
+        }
+
+        return function cleanup() {
+            _observer?.disconnect();
+        };
+    }, [isMounted, anchorElement, autoUpdate]);
+
+    const _renderPositionedContent = (
         popperProps: PopperChildrenProps,
-    ): React.ReactNode {
-        const {children} = this.props;
+    ): React.ReactNode => {
```

Comment:
**suggestion:** I think this needs to be in a `useCallback` otherwise it will cause the Popper to re-render every time since it keeps changing.